modules/channel-slack/src/backend/client.ts

```diff
@@ -17,27 +16,31 @@ const debugOutgoing = debug.sub('outgoing')
 const outgoingTypes = ['text', 'image', 'actions', 'typing', 'carousel']
 
 export class SlackClient {
-  private router: Router & sdk.http.RouterExtension
   private client: WebClient
   private rtm: RTMClient
   private interactive: any
-  private botId: string
+  private logger: sdk.Logger
 
-  constructor(private bp: typeof sdk, botId: string, config: Config, router) {
-    this.botId = botId
-    this.router = router
-
-    this.client = new WebClient(config.botToken)
-    this.rtm = new RTMClient(config.botToken)
-    this.interactive = createMessageAdapter(config.signingSecret, {}) as any
+  constructor(private bp: typeof sdk, private botId: string, private config: Config, private router) {
+    this.logger = bp.logger.forBot(botId)
   }
 
   async initialize() {
+    if (!this.config.botToken || !this.config.signingSecret) {
+      return this.logger.error(
```

Comment:
this should maybe be 2 different instructions ?

this.logger.error(...)
return

Reply:
The logger returns void and it's the same for initialize, so it has no impact here